roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,49 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerized
+
+class DockerImageAvailability(NotContainerized, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+
+        args = {
+            "name": self.rpm_docker_images(),
+        }
+
+        docker_images = self.module_executor("docker_image_facts", args, tmp, task_vars)
+
+        if len(docker_images['images']) == 0:
+            return {"failed": True, "msg": "Unable to find any docker images!"}
+
+        # if we find any docker images, attempt to do 1-to-1 match between
+        # each image description name key and items from rpm_docker_images
+        matched_images = set()
+        for i in docker_images['images']:
+            repoTag = i['RepoTags'][0]
+
+            # check image version
+            tag = repoTag.split(':')
+            if tag[1] == openshift_release and repoTag[0] in self.rpm_docker_images():
+                matched_images.add(tag[0])
+
+        missing_images = set(self.rpm_docker_images()) - matched_images
+        if len(missing_images) > 0:
+            return {"failed": True, "msg": "There are missing docker images or images that did not match the current OpenShift version (%s): %s:\n" % (openshift_release, missing_images)}
+
+        return {"Changed": False}
+
+    @staticmethod
+    def rpm_docker_images():
+        return [
+            "docker.io/openshift/origin",
+            "registry.access.redhat.com/openshift3/ose-haproxy-router",
```

Comment:
As @sdodson said in https://github.com/openshift/openshift-ansible/pull/3461#issuecomment-281880558, we need to support the case when the repository prefix is something else.

Perhaps we can get the "something else" precisely from the variable he pointed us at: `oreg_url`.